### PR TITLE
chore: git refactor first set of apis

### PR DESCRIPTION
**Pull request by sondermanish**
This is a non breaking change. code is not functional yet
fixes: https://github.com/appsmithorg/appsmith/issues/31294

## Summary by CodeRabbit

- **Refactor**
  - Updated method names and parameters across various services to reflect a shift from `ApplicationGitReference` to `ArtifactGitReference`, enhancing the Git integration features.
  - Renamed `ExportableArtifact` and `ImportableArtifact` to `Artifact` to unify artifact handling across export and import services, streamlining the workflow for users.

- **New Features**
  - Introduced `ArtifactGitReference` and `Artifact` interfaces to better manage Git operations and artifact handling, providing a more robust and flexible framework for version control integration.
  - Added new utility functions for Git operations related to applications, improving the efficiency of Git-related tasks such as committing, cloning, and managing artifacts.

- **Bug Fixes**
  - Adjusted logic related to scanning and deleting directories in Git operations to prevent potential issues with artifact management.

- **Documentation**
  - Updated method signatures and references in documentation to align with the new artifact handling approach, ensuring clarity and consistency for developers and users.

### CODE REVIEW

File: app/server/appsmith-git/src/main/java/com/appsmith/git/service/ce/GitExecutorCEImpl.java [hunk 1]
```diff
@@ -235,7 +235,7 @@ public Mono<String> pushApplication(
                 .subscribeOn(scheduler);
     }
 
-    /** Clone the repo to the file path : container-volume/orgId/defaultAppId/repo/applicationData
```
Comment (nidhi-nair) [inline app/server/appsmith-git/src/main/java/com/appsmith/git/service/ce/GitExecutorCEImpl.java:238]: Where's the logic for calculating this path? Gotta split that too right?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/applications/git/GitApplicationHelperCEImpl.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        update.setClientSchemaVersion(JsonSchemaVersions.clientVersion);
+        update.setServerSchemaVersion(JsonSchemaVersions.serverVersion);
+        update.setIsManualUpdate(false);
+
+        return applicationService.update(artifact.getId(), update);
+    }
+
+    @Override
+    public Path getRepoSuffixPath(String workspaceId, String artifactId, String repoName, @NonNull String... args) {
+        List<String> varargs = new ArrayList<>(List.of(artifactId, repoName));
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/applications/git/GitApplicationHelperCEImpl.java:104]: Ah this one, found it :D

File: app/server/appsmith-server/src/main/java/com/appsmith/server/applications/git/GitApplicationHelperCEImpl.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        String defaultPageId = "";
+        if (!application.getPages().isEmpty()) {
+            defaultPageId = application.getPages().stream()
+                    .filter(applicationPage -> applicationPage.getIsDefault().equals(Boolean.TRUE))
+                    .toList()
+                    .get(0)
+                    .getId();
+        }
+
+        String viewModeUrl = Paths.get("/", Entity.APPLICATIONS, "/", artifact.getId(), Entity.PAGES, defaultPageId)
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/applications/git/GitApplicationHelperCEImpl.java:143]: What does this path indicate?

Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/applications/git/GitApplicationHelperCEImpl.java:143]: This path is inlcuded in the readme of the application, from where you can view/open the application 
image

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**